adit-client/adit_client/client.py

```diff
@@ -67,6 +69,200 @@ def retrieve_study(
             additional_params=additional_params,
         )
 
+    def retrieve_nifti_study(self, ae_title: str, study_uid: str) -> list[tuple[str, BytesIO]]:
+        """
+        Retrieve NIfTI files from the API for a specific study.
+
+        Args:
+            ae_title: The AE title of the server.
+            study_uid: The study instance UID.
+
+        Returns:
+            A list of tuples containing the filename and file content.
+        """
+        # Construct the full URL
+        url = f"{self.server_url}/api/dicom-web/{ae_title}/wadors/studies/{study_uid}/nifti"
+
+        # Call the API
+        response = self._create_dicom_web_client(ae_title)._http_get(
+            url,
+            headers={"Accept": "multipart/related; type=application/octet-stream"},
+        )
+
+        # Create a decoder for the multipart response
+        decoder = MultipartDecoder.from_response(response)
+        files = []
+
+        # Process each part in the multipart response
+        for part in decoder.parts:
+            content_disposition = part.headers.get(b"Content-Disposition", b"").decode("utf-8")  # type: ignore
+            filename = self._extract_filename(content_disposition)
+            files.append((filename, BytesIO(part.content)))
+
+        return files
+
+    def iter_nifti_study(self, ae_title: str, study_uid: str) -> Iterator[tuple[str, BytesIO]]:
+        """
+        Iterate over NIfTI files from the API for a specific study.
+
+        Args:
+            ae_title: The AE title of the server.
+            study_uid: The study instance UID.
+
+        Yields:
+            Tuples containing the filename and file content.
+        """
+        # Construct the full URL
+        url = f"{self.server_url}/api/dicom-web/{ae_title}/wadors/studies/{study_uid}/nifti"
+
+        # Call the API
+        response = self._create_dicom_web_client(ae_title)._http_get(
```

Comment:
The reason for the iter functions is to yield the files as soon as they are received. You are still downloading the whole study and only then yield each series one by one. Therefore, this implementation is not superior to the retrieve method.